mesonbuild/build.py

```diff
@@ -634,11 +637,12 @@ def link(self, target):
             if hasattr(t, 'held_object'):
                 t = t.held_object
             if not isinstance(t, (StaticLibrary, SharedLibrary)):
-                raise InvalidArguments('Link target is not library.')
+                raise InvalidArguments('Link target {!r} is not library.'.format(t.name))
             if isinstance(self, SharedLibrary) and isinstance(t, StaticLibrary) and not t.pic:
-                raise InvalidArguments("Can't link a non-PIC static library into a shared library")
+                msg = "Can't link non-PIC static library {!r} into shared library {!r}. ".format(t.name, self.name)
+                msg += "Use the 'pic' option to static_library to build with PIC."
```

Comment:
Although subprojects make this interesting since you don't necessarily control the contents of a subproject, and if you want to link to a static library from a subproject, we might want to enable `pic` automatically.